### DISCUSSION TITLE
Shouldn't the ER Diagram sub-tab be before the Data sub-tab?

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#15641] [Q&A]
I mean, it's not a big deal, but in order of generation, the table properties are made first (CREATE statement), then relations with other tables are made (which attach when those tables are made), then data is inserted... so intuitively, the Properties sub-tab should be first, then the ER Diagram sub-tab, then the Data sub-tab.

![image](https://user-images.githubusercontent.com/74785256/155741287-ece20633-a2fa-4484-adb0-0d08c693092c.png)

However, I can't rearrange these tabs, they're fixed. I wouldn't mind being able to rearrange them, either per table or in settings as a default, but I'm not asking for that.

I just think it would make more sense to have Data after ER Diagram.

**Comment by LonwoLonwo** [comment]
Hello @givihuda 

The order of the tab is defined by usability. 
Our users are much more likely to use the Data Viewer tab and the ability to edit data in their tables than the ER-diagram tab.
Therefore, we do not plan to change the order of the tabs in the near future.

**Reply by givihuda** [reply]
There's a way to address this even when the tabs are in sensible order: have the default tab be Data.

While I don't necessarily agree that the default tab should be Data (considering that could actually slow down the editor or crash it, depending on the table data, which I've seen a few times), I think it'd be an OK solution if users are more likely to look at that tab than Diagram.

My thinking is just that since a table structure is built first (column definitions), that the Diagram tab would be ordered before the Data tab; however, per your usability's sake reasoning, that order actually wouldn't impact usability at all, because they're so small that the amount of moving a mouse pointer is insignificant, and more often than not I reach the Data tab by right-clicking the table name then selecting View Data.

So with my usage, the order of those tabs have zero impact on the app's usability to me; they just appear to be in the wrong order, because table structure happens before data insert. See what I'm saying? I'm not seeing a usability detriment with proper tab order, just a display fix.
